src/amo/pages/Guides/index.js

```diff
@@ -0,0 +1,214 @@
+  constructor(props: InternalProps) {
+    super(props);
+
+    const { guidesSlug, guids } = this.props;
+
+    this.props.dispatch(
+      fetchGuidesAddons({
+        guids,
+        errorHandlerId: guidesSlug,
+      }),
+    );
+  }
+
+  getGuidesSectionsHtml = (
+    sections: Array<Object>,
+  ): React.ChildrenArray<React.Node> => {
+    const { addons, clientApp, i18n, lang } = this.props;
+
+    return addons.length
+      ? addons.map((addon, key) => {
+          // TODO: look into having these links use the Router (vs 'a' tag).
+          // See https://github.com/mozilla/addons-frontend/issues/6787.
+          const exploreMore = i18n.sprintf(sections[key].exploreMore, {
+            linkStart: `<a href="/${lang}/${clientApp}${
+              sections[key].exploreUrl
+            }">`,
+            linkEnd: '</a>',
+          });
+
+          return (
+            <div className="Guides-section" key={addon && addon.url}>
+              <h2 className="Guides-section-title">{sections[key].header}</h2>
+              <p className="Guides-section-description">
+                {sections[key].description}
+              </p>
+
+              {/* eslint-disable react/no-danger */}
+              <div
+                className="Guides-section-explore-more"
+                dangerouslySetInnerHTML={sanitizeHTML(exploreMore, ['a'])}
+              />
+              {/* eslint-enable react/no-danger */}
+            </div>
+          );
+        })
+      : null;
+  };
+
+  getGuidesPage = () => {
+    const { i18n, match } = this.props;
+    const { guidesSlug } = match.params;
+    const { introText, icon, title: pageTitle, sections } = getContent(
+      guidesSlug,
+      i18n,
+    );
+
+    if (!pageTitle) {
+      return null;
+    }
+
+    return (
+      <div className="Guides">
+        <div className="Guides-header">
+          <Icon className="Guides-header-icon" name={icon} />
+          <h1 className="Guides-header-page-title">{pageTitle}</h1>
+          <p className="Guides-header-intro">{introText}</p>
+        </div>
+        {this.getGuidesSectionsHtml(sections)}
+      </div>
+    );
+  };
+
+  render() {
+    const { i18n } = this.props;
+    const guidesPage = this.getGuidesPage();
+
+    if (!guidesPage) {
+      return <NotFound />;
+    }
+
+    const { match } = this.props;
+    const { guidesSlug } = match.params;
+    const { title } = getContent(guidesSlug, i18n);
+    const pageTitle = i18n.sprintf('%(pageTitle)s', {
+      pageTitle: title,
+    });
+
+    return (
+      <div className="Guides-page">
+        <Helmet>
+          <title>{pageTitle}</title>
+        </Helmet>
+        {guidesPage}
+      </div>
+    );
+  }
+}
+
+export const mapStateToProps = (state: AppState, ownProps: Props) => {
+  const { match } = ownProps;
+  const { guidesSlug } = match.params;
+
+  const addons = [];
+  const guids = getGuids(guidesSlug) || [];
+
+  guids.forEach((guid) => {
```

Comment:
I think this should be done in the `render()` method instead, so that you can map each guid to a "section". That would avoid the implicit contract that is: the order of the array of add-ons.

Reply:
but I have to have state to use this reducer: `getAddonByGUID`. Is it okay to pass all of state as a prop?

or let me know if I miss something..

Reply:
oh, I see now. Maybe you can build the map here then, instead of having `guids` as an array, let's have a `{[guid: string]: AddonType }`

Reply:
Another thought I had, is I can just add the addon guid to its respective section and then pass the guid to the `GuidesAddonCard`:

```javascript
 <GuidesAddonCard
     addonGuid={section.addonGuid}
  />
```

Then I can use the `getAddonByGUID` reducer from `GuidesAddonCard`. I am going to try that. Hopefully you will agree : )